### ISSUE TITLE
The product template should contain appropriate landmarks

### ISSUE THREAD

**Issue by JuliaHempel** [zopaUK/react-components#685]
The [product template](https://github.com/zopaUK/react-components/tree/master/src/components/templates/ProductTemplate) should have a `<main>` in there somewhere so we comply with the a11y rule of [Document should have one main landmark](https://www.w3.org/TR/wai-aria-practices/examples/landmarks/main.html#:~:text=Each%20page%20should%20have%20one,may%20have%20one%20main%20landmark)

The back button and progress bar should probably be in a `<section>` landmark
